nixos/doc/manual/development/etc-overlay.section.md

```diff
@@ -0,0 +1,28 @@
+# `/etc` via overlay filesystem {#sec-etc-overlay}
+
+Note: This is experimental and requires a kernel version >= 6.6 because it uses
+new overlay features and relies on the new mount API.
+
+Instead of using a custom perl script to activate `/etc`, you activate it via an
+overlay filesystem:
+
+```nix
+system.etc.overlay.enable = true;
+```
+
+Using an overlay has two benefits:
+
+1. it removes a dependency on perl
+2. it makes activation faster (up to a few seconds)
+
+By default, the `/etc` overlay is mounted writable (i.e. there is a writable
+upper layer). However, you can also mount `/etc` immutably (i.e. read-only) by
+setting:
+
+```nix
+system.etc.overlay.mutable = false;
+```
+
+The overlay is atomically replaced during system switch. However, files that
+have been modified will NOT be overwritten. This is the biggest change compare
+to the perl-based system.
```

Comment:
Does this mean that toggling `mutable` to `true`. Making edits and then toggling to `false` resets `/etc` back to it's original configuration?

Reply:
Yes. I'll add that example to the docs.